Add an attachments section

danielcmccarthy opened the pull request:
Hello,

This PR adds a section listing attachments for the current issue. Pressing `RET` on an attachment downloads it into a buffer. This is based on some code I had using the `jiralib2` package (https://github.com/nyyManni/jiralib2). It only required one change: `jira-api-call` has a new `:parser` keyword arg. This is required because Jira attachment URLS provide raw binary data, instead of JSON like all other Jira APIs.

I also have code for deleting attachments and uploading a buffer as a new attachment. I can add those if you're interested.

Thanks for creating this!

unmonoqueteclea commented:
Very interesting!! Please,  rebase it on top of last main and resolve the conflicts. I will try to review and test it next week

danielcmccarthy commented:
Rebased and resolved. I also added a message before downloading.  Even if `jira-debug` is disabled, downloads can take substantially longer than normal API calls, so the user might want to know what's going on.
